app/services/ZuoraService.scala

```diff
@@ -27,8 +30,69 @@ class ZuoraService(zuoraApiConfig: ZuoraApiConfig) extends ZuoraApi {
-    request(Subscribe(memberId, data, ratePlanId, paymentDelay))
+  def createSubscription(memberId: MemberId, data: SubscriptionData, paymentDelay: Option[Period]): Future[SubscribeResult] = {
+    request(Subscribe(memberId, data, paymentDelay))
   }
+
+  private def getProducts(): Future[Seq[SubscriptionProduct]] = {
+
+    def productRatePlans: Future[Seq[ProductRatePlan]] =
+      query[ProductRatePlan](s"ProductId='$digitalProductId'")
+
+
+    def productRatePlanCharges(productRatePlans: Seq[ProductRatePlan]): Future[Seq[ProductRatePlanCharge]] = {
+      Future.traverse(productRatePlans) {  productRatePlan =>
+          query[ProductRatePlanCharge](s"ProductRatePlanId='${productRatePlan.id}'")
+        }.map(_.flatten)
+      }
+
+
+    def productRatePlanChargeTiers(productRatePlanCharges: Seq[ProductRatePlanCharge]): Future[Seq[ProductRatePlanChargeTier]] = {
+      Future.traverse(productRatePlanCharges) { productRatePlanCharge =>
+        query[ProductRatePlanChargeTier](s"ProductRatePlanChargeId='${productRatePlanCharge.id}'")
+      }.map(_.flatten)
+
+    }
+
+    for {
+      productRatePlan <- productRatePlans
+      productRatePlanCharges <- productRatePlanCharges(productRatePlan)
+      productRatePlanChargesTier <- productRatePlanChargeTiers(productRatePlanCharges)
+    } yield {
+
+      //filter by GBP rate plans
+      val productRatePlanChargesTierPounds = productRatePlanChargesTier.filter(_.currency == "GBP")
+
+      //filter by plans that have a billing frequency we have
+      val validProductRatePlanCharges = productRatePlanCharges.filter(b => BillingFrequency.all.map(_.lowercase).contains(b.billingPeriod.toLowerCase))
+
+      //filter product rate plans by ones that are in date
+      val validRatePlans = productRatePlan.filter { p =>
+        p.effectiveStartDate.isBeforeNow && p.effectiveEndDate.isAfterNow
+      }
+
+      //pull all the queries together to make a SubscriptionProduct
+
+      val subscriptionProducts = for(ratePlan <- validRatePlans) yield {
+        val billingFreqProductRatePlan = validProductRatePlanCharges.find(_.productRatePlanId == ratePlan.id)
+        val productRatePlanChargeTier = billingFreqProductRatePlan.flatMap { p =>
+          productRatePlanChargesTierPounds.find(_.productRatePlanChargeId == p.id)
+        }
+
+        val billingFreq = billingFreqProductRatePlan.flatMap(a => BillingFrequency.all.find(a.billingPeriod.toLowerCase == _.lowercase))
+
+        if (billingFreq.isDefined && productRatePlanChargeTier.isDefined)
+          SubscriptionProduct(Digital, billingFreq.get, ratePlan.id, productRatePlanChargeTier.get.price)
+        else throw new ZuoraServiceError(s"Could not find valid ratePlan details for Digital Pack. Rate plan ID requested ${ratePlan.id}")
+      }
+      subscriptionProducts
+    }
+  }
+
+  val productsTask = ScheduledTask[Seq[SubscriptionProduct]]("Loading rate plans", Nil, 5.seconds, 1.day)(getProducts())
```

Comment:
I haven't looked into this yet but mainly as I want to see if this will work as-is in production. 

We can alter the configuration times, I've set a 5 second delay from the auth request. I've added to the healthcheck that the list of products must be nonEmpty so an instance won't be able to serve requests until it passes this check.